### ISSUE TITLE
descrepancy between gene abundance and tx abundance.

### ISSUE THREAD

**Issue by PoisonAlien** [gpertea/stringtie#36]
Hi, I used stringtie v1.2.1 to estimate FPKM values against ensemble 75 gtf file on an alignment file generated with tophat2.

Stringtie command:

```bash
stringtie accepted_hits.bam -G ~/NGS/gatk_ref/ensembl_75_hg19.gtf -p 8 -A delMeLater/fuddls1.abundance -B -e -x chrM -o delMeLater/fuddls1.op
```

But the values estimated for all transcripts of a gene in transcript abundance file `t_data.ctab` do not add up to the value estimated for the same gene in gene abundance file. 

For example, here is GAPDH expression from `gene.abundance`

```bash
$ grep -w GAPDH fuddls1.abundance 
ENSG00000111640	GAPDH	+	6643093	6647537	3999	7632.947266	697.140625	1722.543457
```

These GAPDH transcript abundances from `t_data.ctab`
```bash
$ grep -w GAPDH t_data.ctab 
36909	chr12	+	6643093	6647537	ENST00000229239	9	1875	ENSG00000111640	GAPDH	0.202568	0.018422
36910	chr12	+	6643678	6644307	ENST00000496049	2	390	ENSG00000111640	GAPDH	9.696434	0.881809
36911	chr12	+	6643698	6647506	ENST00000396856	9	1266	ENSG00000111640	GAPDH	0.271702	0.024709
36912	chr12	+	6643699	6646964	ENST00000492719	8	930	ENSG00000111640	GAPDH	29.442436	2.677542
36913	chr12	+	6643700	6647525	ENST00000396861	9	1348	ENSG00000111640	GAPDH	0.351579	0.031973
36914	chr12	+	6643707	6647480	ENST00000474249	8	1333	ENSG00000111640	GAPDH	0.214894	0.019543
36915	chr12	+	6643920	6647483	ENST00000466588	7	1363	ENSG00000111640	GAPDH	0.344162	0.031299
36916	chr12	+	6643920	6647505	ENST00000396859	8	1256	ENSG00000111640	GAPDH	0.367886	0.033456
36917	chr12	+	6643958	6647503	ENST00000466525	4	1720	ENSG00000111640	GAPDH	0.213218	0.019390
36918	chr12	+	6644468	6647481	ENST00000396858	8	1292	ENSG00000111640	GAPDH	0.357635	0.032524
```
Gene abundance gives a value of 1722.543457 FPKM for GAPDH which is what we expect but if I add up all values of GAPDH tarnscripts (last column FPKM) from t_data.ctab, it adds up to 3.77. 

Is there anything I'm misinterpreting here ? I also used cufflinks to estimate the same, there GAPDH transcript values from `isoforms.fpkm_tracking` adds up to GAPDH gene value in `genes.fpkm_tracking`. 

Thank you.

**Comment by mpertea** [comment]
Hi PoisonAllen,
It would help a great deal if you could provide us with the bam file with the reads in the region chr12 : 6643093 - 6647537 as well as the annotation for all isoforms in GAPDH. We would be very grateful if you could extract these two files for us and we could tell you exactly why this happens.
I believe it is somehow related to the way we create the abundance file that includes the gene expression computed from  all reads that would go to isoforms we assembled plus the reads that would go to isoforms that we know we can not assemble due to too low levels of coverage. We made StringTie 1.2.0 to include partially covered isoforms with the -e parameter but apparently we did not update the gene abundance file computation somehow.
Ela

**Comment by PoisonAlien** [comment]
Hi Ela,
Here are the bam file and gtf file for gapdh locus. 

https://www.dropbox.com/sh/xtusqbu3d9kt8tt/AAB1Y0uJsELiPG8vpQBi_W_Ka?dl=0

Hope this helps.
Thank you.

**Comment by mpertea** [comment]
Got them - thank you! I'll look at them and let you know.

**Comment by mpertea** [comment]
I ran stringtie v1.2.1 on the two files you provide and I couldn't reproduce the problem you mention, so I don't know what is going on. Transcript ENST00000229239 is the most highly expressed, and its FPKM is very close to the gene level expression, not like in your output. When you run stringtie without the -x chrM, does the output in t_data.ctab look the same?

**Comment by PoisonAlien** [comment]
Hi, you are rite. I also used stringtie on this gapdh bam file and fpkm values do correlate. But when I run this on whole bam file, it gives me very different values. I also tried running stringtie without -x chrM but the values still differ for this gene. 

But, I took `t_data.ctab` file and summerized FPKM values by gene_id and correlated with values from gene.abundance. They do correlate very well except for few genes. 

![git](https://cloud.githubusercontent.com/assets/8164062/12636106/da01576a-c5c4-11e5-8503-ac532b25c846.png)

I am sorry that jumped into conclusion just by looking at this one gene.

**Comment by mpertea** [comment]
I still would like to see why this gene behaves differently so it won't happen again. There is no reason in my mind why the expression would be different if you use the whole bam file. Would it be possible for you to give me the input bam and the annotation file so I can look into this? If you don't have the space where to put them, please contact me at mpertea at jhu . edu and I can send you instructions privately of an ftp space you could use.  I appreciate your help!

**Comment by PoisonAlien** [comment]
Hi Mihaela,

No problem. I have some space left in my onedrive. Here is the link to the input bam file and reference gtf used to quantify against. 

https://onedrive.live.com/redir?resid=DBE0896B86115076!808&authkey=!AC5Wdjq1LEEoqVI&ithint=folder%2cbam

Hope this helps. 

Thank you.

**Comment by gpertea** [comment]
We tried to download the files but the bam file gave this error (yeah, "later" too..)
>We're unable to complete your request
>Microsoft account is experiencing technical problems. Please try again later.

We have a ftp site where you could upload that BAM file - could you please send me a quick note at gpertea@jhu.edu to give you the ftp upload info?

**Comment by mpertea** [comment]
Issue was fixed in StringTie version 1.2.2 - just released today.

**Comment by PoisonAlien** [comment]
Hi,

Thank you for fixing the issue. I checked it over multiple bam files, and it works perfect !
